server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+
+        mock_resp.assert_called_once_with(sync_resource.manager.update().for_display())
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch('pulp.server.webservices.views.repositories.generate_json_response')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.manager_factory.repo_sync_schedule_manager')
+    def test_update_sync_schedule_with_schedule_param(self, mock_manager, mock_reverse, mock_resp):
+        """
+        Update a schedueld sync with all required parameters.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({'schedule': 'data'})
+        sync_resource = RepoSyncScheduleResourceView()
+        response = sync_resource.put(mock_request, 'mock_repo', 'mock_importer', 'mock_schedule')
+        sync_resource.manager.update.assert_called_once_with(
+            'mock_repo', 'mock_importer', 'mock_schedule', {'iso_schedule': 'data'})
+
+        mock_resp.assert_called_once_with(sync_resource.manager.update().for_display())
+        self.assertTrue(response is mock_resp.return_value)
+
+
+class TestRepoDistributorsView(unittest.TestCase):
+    """
+    Tests for RepoDistributorsView.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_get_distributors(self, mock_factory, mock_resp):
+        """
+        Get distributors for a repository.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_dist = [{"mock": "distributor"}]
+        mock_factory.repo_distributor_manager.return_value.get_distributors.return_value = mock_dist
+        repo_importers = RepoDistributorsView()
+        response = repo_importers.get(mock_request, 'mock_repo')
+        mock_resp.assert_called_once_with(mock_dist)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch('pulp.server.webservices.views.repositories.generate_redirect_response')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_distributors(self, mock_factory, mock_rev, mock_resp, mock_redir):
+        """
+        Associate a distributor to a repository with minimal options.
```

Comment:
empty body cannot be minimal options here, because in this call there are some required params that cannot be omitted. http://fpaste.org/203748/27459187/